src/wallet/wallet_rpc_server.cpp

```diff
@@ -3891,7 +3891,20 @@ namespace tools
       er.message = "Failed to parse view key secret key";
       return false;
     }
-
+    
+    crypto::public_key pkey;
+    if (!crypto::secret_key_to_public_key(viewkey, pkey)) {
+          er.code = WALLET_RPC_ERROR_CODE_UNKNOWN_ERROR;
+          er.message = "failed to verify view key secret key";
+          return false;
+        }
+        
+    if (info.address.m_view_public_key != pkey) {
+      er.code = WALLET_RPC_ERROR_CODE_UNKNOWN_ERROR;
+      er.message = "view key does not match standard address";
+      return false;
+    }
```

Comment:
```suggestion
    hw::device &hwdev = hw::get_device("default");
    if (!hwdev.verify_keys(viewkey, info.address.m_view_public_key)) {
      er.code = WALLET_RPC_ERROR_CODE_UNKNOWN_ERROR;
      er.message = "view secret key does not match main address";
      return false;
    }
```

ACK the idea. IMO this is a more succinct and clearer way to write the check. Also, the term "main addresses" should used instead of "standard address". Same comments go for the spend privkey check below.